app/services/fees/charge_service.rb

```diff
@@ -2,12 +2,14 @@
 
 module Fees
   class ChargeService < BaseService
-    def initialize(invoice:, charge:, subscription:, boundaries:)
+    def initialize(invoice:, charge:, subscription:, boundaries:, currency: nil)
       @invoice = invoice
       @charge = charge
       @subscription = subscription
       @is_current_usage = false
       @boundaries = OpenStruct.new(boundaries)
+      @currency = currency || invoice.total_amount.currency
```

Comment:
instead of adding a new argument here, you can probably use `charge.plan.amount_curency`